velox/expression/tests/ArgumentTypeFuzzerTest.cpp

```diff
@@ -403,4 +403,205 @@ TEST_F(ArgumentTypeFuzzerTest, orderableConstraint) {
+
+  argTypes = fuzzArgumentTypes(*signature, DECIMAL(10, 7));
+  ASSERT_EQ(4, argTypes.size());
+  EXPECT_EQ(DECIMAL(10, 7)->toString(), argTypes[0]->toString());
+  EXPECT_EQ(DECIMAL(10, 10)->toString(), argTypes[1]->toString());
+  EXPECT_EQ(DECIMAL(12, 7)->toString(), argTypes[2]->toString());
+  EXPECT_EQ(DECIMAL(2, 1)->toString(), argTypes[3]->toString());
+}
+
+TEST_F(ArgumentTypeFuzzerTest, fuzzDecimalReturnType) {
+  auto fuzzReturnType = [](const exec::FunctionSignature& signature) {
+    std::mt19937 seed{0};
+    ArgumentTypeFuzzer fuzzer{signature, seed};
+    return fuzzer.fuzzReturnType();
+  };
+
+  // Return type can be any decimal.
+  auto signature = exec::FunctionSignatureBuilder()
+                       .integerVariable("p")
+                       .integerVariable("s")
+                       .returnType("decimal(p,s)")
+                       .argumentType("decimal(p,s)")
+                       .build();
+
+  auto returnType = fuzzReturnType(*signature);
+  EXPECT_TRUE(returnType->isDecimal());
+
+  // Return type can be any decimal with scale 3.
+  signature = exec::FunctionSignatureBuilder()
+                  .integerVariable("p")
+                  .integerVariable("s")
+                  .returnType("decimal(p,3)")
+                  .argumentType("decimal(p,s)")
+                  .build();
+
+  returnType = fuzzReturnType(*signature);
+  EXPECT_TRUE(returnType->isDecimal());
+  EXPECT_EQ(3, getDecimalPrecisionScale(*returnType).second);
+
+  // Another way to specify that scale must be 3.
+  signature = exec::FunctionSignatureBuilder()
+                  .integerVariable("p")
+                  .integerVariable("s", "3")
+                  .returnType("decimal(p,s)")
+                  .argumentType("decimal(p,s)")
+                  .build();
+
+  returnType = fuzzReturnType(*signature);
+  EXPECT_TRUE(returnType->isDecimal());
+  EXPECT_EQ(3, getDecimalPrecisionScale(*returnType).second);
+
+  // Return type can be any decimal with precision 22.
+  signature = exec::FunctionSignatureBuilder()
+                  .integerVariable("p")
+                  .integerVariable("s")
+                  .returnType("decimal(22,s)")
+                  .argumentType("decimal(p,s)")
+                  .build();
+
+  returnType = fuzzReturnType(*signature);
+  EXPECT_TRUE(returnType->isDecimal());
+  EXPECT_EQ(22, getDecimalPrecisionScale(*returnType).first);
+
+  // Another way to specify that precision must be 22.
+  signature = exec::FunctionSignatureBuilder()
+                  .integerVariable("p", "22")
+                  .integerVariable("s")
+                  .returnType("decimal(p,s)")
+                  .argumentType("decimal(p,s)")
+                  .build();
+
+  returnType = fuzzReturnType(*signature);
+  EXPECT_TRUE(returnType->isDecimal());
+  EXPECT_EQ(22, getDecimalPrecisionScale(*returnType).first);
+
+  // Return type can only be DECIMAL(10, 7).
+  signature = exec::FunctionSignatureBuilder()
```

Comment:
can you add a test case where there are multiple variables for precision or scale like `decimal(p1,s1), decimal(p2,s2)`

Reply:
@bikramSingh91 Happy to add more test cases, but would you clarify a bit more of what kind of test you have in mind?

Reply:
Just a sanity check that it handles cases where the signature has more than one precision and scale integer variables like:
```
  auto signature = exec::FunctionSignatureBuilder()
                       .integerVariable("p1")
                       .integerVariable("s1")
                       .integerVariable("p2")
                       .integerVariable("s2")
                       .returnType("boolean")
                       .argumentType("decimal(p1,s1)")
                       .argumentType("decimal(p2,s2)")
                       .build();
```

Reply:
@bikramSingh91 Got it. Added

```
  // Multiple pairs of precision and scale variables.
  signature = exec::FunctionSignatureBuilder()
                  .integerVariable("p1")
                  .integerVariable("s1")
                  .integerVariable("p2")
                  .integerVariable("s2")
                  .returnType("bigint")
                  .argumentType("decimal(p1,s1)")
                  .argumentType("decimal(p2,s2)")
                  .argumentType("decimal(p1,s1)")
                  .argumentType("decimal(p2,s2)")
                  .build();
  argTypes = fuzzArgumentTypes(*signature, BIGINT());
  ASSERT_EQ(4, argTypes.size());
  EXPECT_TRUE(argTypes[0]->isDecimal());
  EXPECT_TRUE(argTypes[1]->isDecimal());
  EXPECT_EQ(argTypes[0]->toString(), argTypes[2]->toString());
  EXPECT_EQ(argTypes[1]->toString(), argTypes[3]->toString());
```